code/game/objects/structures/omnius.dm

```diff
@@ -0,0 +1,67 @@
+/obj/structure/omnius
+	name = "omnius generator"
+	icon_state = "omnius"
+	icon = 'icons/obj/machines/excelsior/objects.dmi'
+	var/cooldown = 0
+
+	attack_hand(mob/living/user as mob)
+		if(cooldown == 0)
+			cooldown = 1
+			emp_in(src.loc, 1, 1, 0)
+			sleep(460)
```

Comment:
That's hell of a sleep. Maybe just check world time?
e.g.
var/last_user

if(world.time < last_use + 46 SECONDS)
  return
last_use = world.time
emp_in(src.loc, 1, 1, 0)
```